internal/server/repository.go

```diff
@@ -0,0 +1,137 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"sort"
+	"strings"
+
+	"github.com/osbuild/image-builder/internal/cloudapi"
+)
+
+type DNFError struct {
+	Kind   string `json:"kind"`
+	Reason string `json:"reason"`
+}
+
+func (err *DNFError) Error() string {
+	return fmt.Sprintf("DNF error occured: %s: %s", err.Kind, err.Reason)
+}
+
+type Repo struct {
+	Baseurl    *string `json:"baseurl,omitempty"`
+	Metalink   *string `json:"metalink,omitempty"`
+	Mirrorlist *string `json:"mirrorlist,omitempty"`
+	Rhsm       bool    `json:"rhsm"`
```

Comment:
We need the container to be subscribed now, I guess?

Reply:
ah dang you're right.